### ISSUE TITLE
Allow more autocomplete values

### ISSUE THREAD

**Issue by fchristant** [validator/validator#945]
I'm using autocomplete="username" on my login form, this triggers a validation error:

_Error: Bad value username for attribute autocomplete on element input: The autofill field name username is not allowed in this context._

I believe this error to be incorrect. autocomplete="username" is a security best practice, is documented on MDN as a valid value, and listed in both HTML specs as a valid value:

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete
https://www.w3.org/TR/html52/sec-forms.html#autofilling-form-controls-the-autocomplete-attribute
https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofilling-form-controls:-the-autocomplete-attribute

**Comment by sideshowbarker** [comment]
The checker supports `autocomplete="username"` in conformance with HTML spec requirements.

For example, the checker reports no error for the followin:

```html
<input type="text" name="email" autocomplete="username">
``` 

So if you have a document for which the checker is reporting *“The autofill field name username is not allowed in this context”*, then please either provide a URL for the document, or else post a snippet of the document that can be used to reproduce the error.

**Comment by fchristant** [comment]
This is the line that triggers the error:

`<input type="email" name="email" id="email" value="" placeholder="Email address" class="c_login__input f_redraw f_form__field-input f_form__field-input--text" autocomplete="username">↩`

It looks like type="email" was the problem. So my bad, sorry!

It seems easy to make this mistake though. For many login systems, email is the username and the incorrect combination I used worked perfectly with both first party and 3rd party password managers, so it seemed ok.

Thanks for help, feel free to close.

**Comment by sideshowbarker** [comment]
Cheers — glad you got it resolved

> It seems easy to make this mistake though. For many login systems, email is the username and the incorrect combination I used worked perfectly with both first party and 3rd party password managers, so it seemed ok.

Yeah, maybe the spec should be updated to allow that…
